content/topics/kotlin/in-line-functions/_index.md

```diff
@@ -0,0 +1,143 @@
+---
+title: In Line Functions
+ready: true
+prerequisites:
+  hard: ["topics/kotlin/functions"]
+  soft: []
+---
+
+Using __higher-order functions__ imposes certain runtime penalties: each function is an object, and it captures a closure, i.e. those variables that are accessed in the body of the function. Memory allocations (both for function objects and classes) and virtual calls introduce runtime overhead.
+
+But it appears that in many cases this kind of overhead can be eliminated by inlining the lambda expressions. The functions shown below are good examples of this situation. I.e., the lock() function could be easily inlined at call-sites. Consider the following case:
```

Comment:
functions shoulkd be inline code for readability and cos they can break things

Reply:
Done